File: sdk/go/examples/basic_usage.go
package main

import (
	"fmt"
	"log"
	"time"

	"github.com/voocel/literedis"
)

func main() {
	// 创建连接池
	pool := literedis.NewPool("localhost:6379", 10, 5, 30*time.Second)
	defer pool.Close()

	// 从连接池获取客户端
	client, err := pool.Get()
	if err != nil {
		log.Fatalf("Failed to get client: %v", err)
	}
	defer pool.Put(client)

	// 字符串操作
	if err := client.Set("key1", "value1", 0); err != nil {
		log.Fatalf("Failed to set key1: %v", err)
	}

	value, err := client.Get("key1")
	if err != nil {
		log.Fatalf("Failed to get key1: %v", err)
	}
	fmt.Printf("key1: %s\n", value)

	// 设置带过期时间的键
	if err := client.Set("key2", "value2", 5*time.Second); err != nil {
		log.Fatalf("Failed to set key2: %v", err)
	}

	ttl, err := client.TTL("key2")
	if err != nil {
		log.Fatalf("Failed to get TTL of key2: %v", err)
	}
	fmt.Printf("TTL of key2: %v\n", ttl)

	// 自增操作
	if err := client.Set("counter", "10", 0); err != nil {
		log.Fatalf("Failed to set counter: %v", err)
	}

	newValue, err := client.Incr("counter")
	if err != nil {
		log.Fatalf("Failed to increment counter: %v", err)
	}
	fmt.Printf("Incremented counter: %d\n", newValue)

	// 删除操作
	deleted, err := client.Del("key1", "key2")
	if err != nil {
		log.Fatalf("Failed to delete keys: %v", err)
	}
	fmt.Printf("Deleted %d key(s)\n", deleted)

	// 批量操作示例
	if err := client.MSet(map[string]interface{}{
		"batch_key1": "batch_value1",
		"batch_key2": "batch_value2",
	}); err != nil {
		log.Fatalf("Failed to set batch keys: %v", err)
	}

	values, err := client.MGet("batch_key1", "batch_key2", "nonexistent_key")
	if err != nil {
		log.Fatalf("Failed to get batch keys: %v", err)
	}
	fmt.Printf("Batch get results: %v\n", values)
}

File: sdk/rust/src/examples/basic_usage.rs
use literedis::{Pool, RedisError, Commands};
use std::time::Duration;
use tokio;

#[tokio::main]
async fn main() -> Result<(), RedisError> {
    // 创建连接池
    let pool = Pool::new("127.0.0.1:6379", 10).await?;
    
    // 从连接池获取连接
    let mut conn = pool.get().await?;
    
    // 字符串操作
    conn.set("key1", "value1").await?;
    let value = conn.get("key1").await?;
    println!("key1: {:?}", value);

    // 设置带过期时间的键
    conn.set_ex("key2", "value2", Duration::from_secs(5)).await?;
    
    // 获取TTL
    let ttl = conn.ttl("key2").await?;
    println!("TTL of key2: {} seconds", ttl);

    // 自增操作
    conn.set("counter", "10").await?;
    let new_value = conn.incr("counter").await?;
    println!("Incremented counter: {}", new_value);

    // 删除操作
    let deleted = conn.del(&["key1", "key2"]).await?;
    println!("Deleted {} key(s)", deleted);

    // 批量操作
    conn.set("batch_key1", "batch_value1").await?;
    conn.set("batch_key2", "batch_value2").await?;
    let values = conn.mget(&["batch_key1", "batch_key2", "nonexistent_key"]).await?;
    println!("Batch get results: {:?}", values);

    // 列表操作
    conn.lpush("mylist", &["item1", "item2", "item3"]).await?;
    let list_length = conn.llen("mylist").await?;
    println!("List length: {}", list_length);
    let list_items = conn.lrange("mylist", 0, -1).await?;
    println!("List items: {:?}", list_items);

    // 哈希表操作
    conn.hset("myhash", "field1", "value1").await?;
    conn.hset("myhash", "field2", "value2").await?;
    let hash_value = conn.hget("myhash", "field1").await?;
    println!("Hash value for field1: {:?}", hash_value);
    let all_fields = conn.hgetall("myhash").await?;
    println!("All hash fields: {:?}", all_fields);

    Ok(())
}
